stl/inc/span

```diff
@@ -153,6 +169,16 @@ struct _Span_iterator {
         return _Myptr == _Right._Myptr;
     }
 
+#if defined(__cpp_impl_three_way_comparison) && __cpp_impl_three_way_comparison >= 201902L
+    _NODISCARD constexpr strong_ordering operator<=>(const _Span_iterator& _Right) const noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        _STL_VERIFY(
+            _Mybegin == _Right._Mybegin && _Myend == _Right._Myend, "cannot compare incompatible span iterators");
+#endif // _ITERATOR_DEBUG_LEVEL >= 1
+        return _Myptr <=> _Right._Myptr;
+    }
+
+#else // ^^^ use spaceship / don't spaceship vvv
     _NODISCARD constexpr bool operator!=(const _Span_iterator& _Right) const noexcept {

```

Comment:
I think this one can be burned anyway?!

Reply:
I'm leaving this here for (admittedly partial) clang support; clang 9 has no spaceship/rewrites. All of the "no spaceship" stuff should disappear when clang 10 is released (assuming LLVM-44627 gets fixed).

Reply:
In that case wouldnt we have to add all the glorious overloads with the different combinations or do you upgrade the requirement on clang 10.

BTW it seems intellisense (EDG I assume) does not get the rewrite for `operator==` or `operator!=` at least it shows me errors for those in the current visual studio preview

Reply:
> In that case wouldnt we have to add all the glorious overloads with the different combinations or do you upgrade the requirement on clang 10.

We bump our minimum required compiler versions aggressively. The rule of thumb is that we require the versions of MSVC/clang that shipped in the most recent Visual Studio Preview. When the VS installer bumps from clang 9 to clang 10, we'll bump our internal testing to clang 10. When the corresponding version of VS ships, we'll bump the required version in `yvals_core.h` and strip out code that's only needed to support an older compiler version.


> BTW it seems intellisense (EDG I assume) does not get the rewrite for `operator==` or `operator!=` at least it shows me errors for those in the current visual studio preview

Reply:
> BTW it seems intellisense (EDG I assume) does not get the rewrite for `operator==` or `operator!=` at least it shows me errors for those in the current visual studio preview

Weird - we test with (what I believe to be) the same version of the IntelliSense engine that ships in VS. (Which is a flavor of EDG's front end, yes.) Are you sure the language mode is set to latest?